src/precice/tests/DataContextTest.cpp

```diff
@@ -55,23 +55,76 @@ BOOST_AUTO_TEST_CASE(testDataContextWriteMapping)
   BOOST_TEST(fixture.getProvidedDataID(dataContext) == ptrFromData->getID());
   BOOST_TEST(dataContext.getMeshID() == ptrFromMesh->getID());
 
-  dataContext.configureMapping(mappingContext, toMeshContext);
+  // Add the first mapping we configured for this context
+  dataContext.addMappingConfiguration(mappingContext, toMeshContext1);
 
   // mapping is configured. Write mapping, therefore _providedData == _fromData
   BOOST_TEST(fixture.hasMapping(dataContext));
-  BOOST_TEST(fixture.getFromDataID(dataContext) == ptrFromData->getID());
-  BOOST_TEST(fixture.getToDataID(dataContext) == ptrToData->getID());
   BOOST_TEST(fixture.getProvidedDataID(dataContext) != ptrToData->getID());
   BOOST_TEST(fixture.getProvidedDataID(dataContext) == ptrFromData->getID());
   BOOST_TEST(dataContext.getMeshID() != ptrToMesh->getID());
   BOOST_TEST(dataContext.getMeshID() == ptrFromMesh->getID());
   BOOST_TEST(fixture.hasWriteMapping(dataContext));
   BOOST_TEST(!fixture.hasReadMapping(dataContext));
-  BOOST_TEST(fixture.mappingContext(dataContext).fromMeshID == mappingContext.fromMeshID);
-  BOOST_TEST(fixture.mappingContext(dataContext).toMeshID == mappingContext.toMeshID);
-  BOOST_TEST(fixture.mappingContext(dataContext).hasMappedData == mappingContext.hasMappedData);
-  BOOST_TEST(fixture.mappingContext(dataContext).mapping == mappingContext.mapping);
-  BOOST_TEST(fixture.mappingContext(dataContext).timing == mappingContext.timing);
+  // Test dedicated content of the first mapping configuration
+  BOOST_TEST(fixture.getFromDataID(dataContext, 0) == ptrFromData->getID());
+  BOOST_TEST(fixture.getToDataID(dataContext, 0) == ptrToData->getID());
+  BOOST_TEST(fixture.mappingContext(dataContext)[0].fromMeshID == mappingContext.fromMeshID);
+  BOOST_TEST(fixture.mappingContext(dataContext)[0].toMeshID == mappingContext.toMeshID);
+  BOOST_TEST(fixture.mappingContext(dataContext)[0].hasMappedData == mappingContext.hasMappedData);
+  BOOST_TEST(fixture.mappingContext(dataContext)[0].mapping == mappingContext.mapping);
+  BOOST_TEST(fixture.mappingContext(dataContext)[0].timing == mappingContext.timing);
+
+  // Add a second mapping targeting a different to mesh
```

Comment:
Can you put this into an individual test? This keeps the existing test simple. Each unit test should also have a single purpose.